sky/adaptors/azure.py

```diff
@@ -38,7 +42,10 @@ def exceptions():
 
 @functools.lru_cache()
 @common.load_lazy_modules(modules=_LAZY_MODULES)
-def get_client(name: str, subscription_id: str):
+def get_client(name: str,
+               subscription_id: str,
+               storage_account_name: Optional[str] = None,
+               container_name: Optional[str] = None):
```

Comment:
Great point. I replaced those two arguments with keyword arguments, and updated `create_az_client`, which calls this function, `get_client`, to do so as well. But I'm hesitant to pass the `container_url` from the function caller. I think it's easier to maintain if the function can construct `container_url` where it actually needs to use it. Less burden to callers and callee both.

Reply:
I dont think this matches the meaning of this function though. This function should only be used to construct an Azure client from arguments?

Reply:
Maybe we could add another helper function to take `storage_account_name` and `container_name` as argument, and return an container url though

Reply:
I updated so that the caller of `create_az_client` obtains the `container_url `and passes it as an argument to the function as you requested. Also, I first wrote a helper function to create the container url, but the function seemed to be shallow, so I wrote a constant format string instead to obtain `container_url`. [d63fc7a](https://github.com/skypilot-org/skypilot/pull/3032/commits/d63fc7a5b757c966a6705357104c53ea8939ac53)

Reply:
This looks great!